aws/resource_aws_network_interface_sg_attachment.go

```diff
@@ -0,0 +1,181 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNetworkInterfaceSGAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNetworkInterfaceSGAttachmentCreate,
+		Read:   resourceAwsNetworkInterfaceSGAttachmentRead,
+		Delete: resourceAwsNetworkInterfaceSGAttachmentDelete,
+		Schema: map[string]*schema.Schema{
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsNetworkInterfaceSGAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	mk := "network_interface_sg_attachment_" + d.Get("network_interface_id").(string)
+	awsMutexKV.Lock(mk)
+	defer awsMutexKV.Unlock(mk)
```

Comment:
Yeah, these are on the outer CRUD functions because I really don't want any dependent read operations getting messed up by concurrency (there are possibly ones in the create/delete operations too).

With that said, I think the overly functional nature of this code in its current form (again a relic of when I had this supporting instances too), so I think on the next refactor run I will probably just flatten most of this functionality to where it makes sense (at the very least, on `attachSecurityGroupToInterface` and `detachSecurityGroupFromInterface`).